kive/container/management/commands/runcontainer.py

```diff
@@ -177,17 +176,25 @@ def build_command(cls, run):
         return command
 
     @staticmethod
-    def _format_kw_arg(dataset: ContainerDataset,
-                       arg: ContainerArgument) -> typing.List[str]:
-        if arg.type == ContainerArgument.INPUT:
-            datasetfolder = "/mnt/input"
-        else:
-            datasetfolder = "/mnt/output"
-        datasetpath = os.path.join(datasetfolder, dataset.name)
-        return [
-            arg.formatted,
-            datasetpath,
-        ]
+    def _format_kw_args(
+            datasets: typing.List[ContainerDataset]) -> typing.Iterable[str]:
+        def sort_by_position(
```

Comment:
Would the sorting and grouping be easier to follow if you used the database query to sort the results on line 161?

Reply:
I think that would let you use `groupby()` here. Think about it, and choose whichever you prefer.

Reply:
I kept this in the application code for two reasons:

1. It's much easier to test this way (not depending on the database means the test code can use mocks instead of persisted `ContainerDataset` objects).
2. I felt that it obfuscated the intent a bit to have the logic for keyword and fixed arguments mixed together, rather than getting the datasets, splitting them into sets, and handling the sets independently.

I also figured that the parameter lists will be small enough that handling them in memory isn't a problem.

That said, I tried the database version and (with the caveat that some tests failed due the above-mentioned difficulty) it did get rid of some of the gnarly sorting code, so in the future I would probably do well to at least try to lean on the database a bit more instead of reaching for dict and list comprehensions quite so quickly.